{
  try {
    closeClient();
  }
 catch (  IOException x) {
    _logger.debug(this + ": unexpected exception closing the client",x);
  }
  try {
    closeServer();
  }
 catch (  IOException x) {
    _logger.debug(this + ": unexpected exception closing the server",x);
  }
}
